src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
Yeah, checked exceptions - I love them ;)

But what I meant by my objection against `Optional` is that I'd like to use it as a parameter (`Optional<T> seed` - solution 2). Actually, when now I look at all these 3 potential solutions, I'm most attracted to this `Optional`-parameter solution. And we already have `Optional` as parameter type, e.g. in `Seq.seq(Optional)`. What do you think?

PS. I don't know about Ceylon and its union types but with Kotlin it would be really nice - we'd have:
```kotlin
fun <T> iterateWhilePresent(seed: T?, generator: (T) -> T?): T
```